fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SplitHelper.java

```diff
@@ -673,6 +685,14 @@ public KeyValueUnsplitter(@Nonnull FDBRecordContext context, @Nonnull final Subs
                         nextVersion = null;
                         nextPrefix = null;
                         nextResult =  RecordCursorResult.withNextValue(result, continuation);
+                        if (LOGGER.isTraceEnabled()) {
+                            KeyValueLogMessage msg = KeyValueLogMessage.build("unsplitter assembled new record",
+                                    LogMessageKeys.NEXT_CONTINUATION, continuation == null ? "null" : ByteArrayUtil2.loggable(continuation.toBytes()),
+                                    LogMessageKeys.KEY_TUPLE, result.getPrimaryKey(),
+                                    LogMessageKeys.SUBSPACE, ByteArrayUtil2.loggable(subspace.getKey()));
+                            result.addSizeLogInfo(msg);
```

Comment:
Didn't we emit a log record when in the append code when we completed the key?

Reply:
Yeah. There should be size information included essentially in the final append, but also including it here means that one doesn't have to correlate the two messages if one wants to see what the size of all the returned records are.